dpctl-capi/source/dpctl_sycl_device_interface.cpp

```diff
@@ -612,11 +613,22 @@ DPCTLDevice_CreateSubDevicesByCounts(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     vector_class<DPCTLSyclDeviceRef> *Devices = nullptr;
     std::vector<size_t> vcounts;
     vcounts.assign(counts, counts + ncounts);
```

Comment:
Since you know the size, might as well reserve the memory saves a realloc.

Reply:
done.